common/test/acceptance/tests/lms/test_problem_types.py

```diff
@@ -0,0 +1,752 @@
+    this class:
+        problem_name (str)
+        problem_type (str)
+        factory (ResponseXMLFactory subclass instance)
+
+    Additionally, the default values for factory_kwargs and status_indicators
+    may need to be overridden for some problem types.
+    """
+    __metaclass__ = ProblemTypeTestBaseMeta
+
+    problem_name = None
+    problem_type = None
+    factory = None
+    factory_kwargs = {}
+    status_indicators = {
+        'correct': ['span.correct'],
+        'incorrect': ['span.incorrect'],
+        'unanswered': ['span.unanswered'],
+    }
+
+    def setUp(self):
+        """
+        Visits courseware_page and defines self.problem_page.
+        """
+        super(ProblemTypeTestBase, self).setUp()
+        self.courseware_page.visit()
+        self.problem_page = ProblemPage(self.browser)
+
+    def get_problem(self):
+        """
+        Creates a {problem_type} problem
+        """
+        # Generate the problem XML using capa.tests.response_xml_factory
+        return XBlockFixtureDesc(
+            'problem',
+            self.problem_name,
+            data=self.factory.build_xml(**self.factory_kwargs),
+            metadata={'rerandomize': 'always'}
+        )
+
+    def wait_for_status(self, status):
+        """
+        Waits for the expected status indicator.
+
+        Args:
+            status: one of ("correct", "incorrect", "unanswered)
+        """
+        msg = "Wait for status to be {}".format(status)
+        selector = ', '.join(self.status_indicators[status])
+        self.problem_page.wait_for_element_visibility(selector, msg)
+
+    @abstractmethod
+    def answer_problem(self, correct):
+        """
+        Args:
+            `correct` (bool): Inputs correct answer if True, else inputs
+                incorrect answer.
+        """
+        raise NotImplementedError()
+
+
+class ProblemTypeTestMixin(object):
+    """
+    Test cases shared amongst problem types.
+    """
+    @attr('shard_2')
+    def test_answer_correctly(self):
+        """
+        Scenario: I can answer a problem correctly
+        Given External graders respond "correct"
+        And I am viewing a "<ProblemType>" problem
+        When I answer a "<ProblemType>" problem "correctly"
+        Then my "<ProblemType>" answer is marked "correct"
+        And The "<ProblemType>" problem displays a "correct" answer
+        And a "problem_check" server event is emitted
+        And a "problem_check" browser event is emitted
+        """
+        # Make sure we're looking at the right problem
+        self.assertEqual(self.problem_page.problem_name, self.problem_name)
+
+        # Answer the problem correctly
+        self.answer_problem(True)
+        self.problem_page.click_check()
+        self.wait_for_status('correct')
+
+        # Check for corresponding tracking event
+        expected_events = [
+            {'event_source': 'server', 'event_type': 'problem_check'},
```

Comment:
You should also filter for username (or something else unique to the user). That will make this resilient for multiprocess testing.